test/functional/p2p_ibd_txrelay.py

```diff
@@ -22,7 +49,40 @@ def set_test_params(self):
             ["-minrelaytxfee={}".format(NORMAL_FEE_FILTER)],
         ]
 
-    def run_test(self):
+    def test_inv_ibd(self):
+        self.log.info("Check that nodes don't request txn while still in IBD")
+        peer = self.nodes[0].add_p2p_connection(P2PInterface())
+        peer.send_message(msg_inv([CInv(t=MSG_WTX, h=0xffaa)]))
+        # As tx-announcing connection is inbound, wait for the inbound delay applied by requester.
+        sleep(NONPREF_PEER_TX_DELAY)
+        peer.sync_with_ping()
+        with p2p_lock:
+            assert_equal(peer.tx_getdata_count, 0)
+        peer.peer_disconnect()
+        peer.wait_for_disconnect()
+
+    def test_unrequested_tx_ibd(self):
+        self.log.info("Check that nodes don't process unrequested txn while still in IBD")
+
+        witness_program = CScript([OP_TRUE])
+        witness_hash = sha256(witness_program)
+        script_pubkey = CScript([OP_0, witness_hash])
+
+        unrequested_tx = CTransaction()
+        unrequested_tx.vin.append(CTxIn(outpoint=COutPoint(uint256_from_str(b"23c70ed7c0506e9178fc1a987f40a33946d4ad4c962b5ae3a52546da53af0c5c"),
+ 0)))
+        unrequested_tx.vout.append(CTxOut(nValue=50 * COIN - 12000, scriptPubKey=script_pubkey))
+        unrequested_tx.calc_sha256()
+
+        peer = self.nodes[0].add_p2p_connection(P2PInterface())
+        peer.send_message(msg_tx(unrequested_tx))
```

Comment:
```suggestion
        peer.send_and_ping(msg_tx(unrequested_tx))
```

Otherwise you are using the debug.log to synchronize. And the `assert_equal(len(self.nodes[0].getrawmempool()), 0)` doesn't actually check anything.

Reply:
I introduced a `sync_with_ping` inside the with statement. Test is breaking as expected with new code muted. Let me know if it achieves what you intended.